aldryn_search/views.py

```diff
@@ -63,6 +63,7 @@ def get_queryset(self):
         queryset = self.form.search()
         if not self.request.user.is_authenticated():
             queryset = queryset.exclude(login_required=True)
+        queryset = queryset.exclude(url__in=['', None])
```

Comment:
aldryn_search always fill url field: https://github.com/aldryn/aldryn-search/blob/master/aldryn_search/base.py#L138

The haystack/aldryn-search does not allow us to ignore objects without urls when building the index (rebuild index, update_index) but only when a object is updated (should_update method on index), so filter by empty urls is the way to do that.